kernel-crawler/minikube-crawler.py

```diff
@@ -0,0 +1,110 @@
+#! /usr/bin/env python3
+
+# This crawler is loosely based on falco's crawler for minikube
+# You can find it here:
+# https://github.com/falcosecurity/kernel-crawler/blob/e2bbe6455ef26941e3f53f9f6481e7a610746484/kernel_crawler/minikube.py
+
+import tempfile
+import sys
+import os
+import re
+import semver
+import pygit2
+
+
+def clone_repo():
+    work_dir = tempfile.mkdtemp(prefix="minikube-")
+    return pygit2.clone_repository("https://github.com/kubernetes/minikube.git", work_dir)
+
+
+def filter_versions(version):
+    if semver.compare('1.24.0', version) > 0:
+        return False
+
+    semver_version = semver.VersionInfo.parse(version)
+    return semver_version.prerelease is None and semver_version.build is None
+
+
+def get_versions(repo):
+    re_tags = re.compile('^refs/tags/v(?P<major>0|[1-9]\d*)\.(?P<minor>0|[1-9]\d*)\.(?P<patch>0|[1-9]\d*)$')
+
+    all_versions = [os.path.basename(v).strip('v') for v in repo.references if re_tags.match(v)]
+    filtered_versions = list(filter(filter_versions, all_versions))
+
+    return filtered_versions
+
+
+def get_minikube_config_file_name(version):
+    if semver.compare('1.26.0', version) <= 0:
+        return 'minikube_x86_64_defconfig'
+    return 'minikube_defconfig'
+
+
+def get_kernel_config_file_name(version):
+    if semver.compare('1.26.0', version) <= 0:
+        return 'linux_x86_64_defconfig'
+    return 'linux_defconfig'
+
+
+def search_files(directory, file_name):
+    for dirpath, dirnames, files in os.walk(directory, followlinks=True):
+        for name in files:
+            if name == file_name:
+                return os.path.join(dirpath, name)
+    return None
```

Comment:
[nit] consider using path lib:

```suggestion
    import pathlib
    return next(pathlib.Path(directory).rglob(file_name), None)
```